Intro/README.md

```diff
@@ -0,0 +1,455 @@
+# Introduction to TPMs
+
+Trusted Platform Modules (TPMs) are a large and complex topic, made all
+the more difficult to explain by the intricate relationships between the
+relevant concepts.  This is an attempt at a simple explanation --
+simpler than reading hundreds of pages of documents, but then too, too
+light on detail to be immediately useful.
+
+So what is a TPM?  Well, it's a cryptographic co-processor with special
+features to enable "root of trust measurement" (RTM), remote attestation
+of system state, unlocking of local resources that are kept encrypted
+(e.g., filesystems), and more.  A TPM can do those things, and it can do
+it with rich authentication and authorization policies.
+
+Typically a TPM is a hardware module, a chip, though there are firmware,
+virtual, and simulated TPMs as well, all implemented in software.
+
+To simplify things we'll consider only TPM 2.0.  Also to simplify things
+we'll ignore algorithm agility.
+
+Other parts of this [tutorial](README.md) may cover specific concepts in
+much more detail.
+
+# Core Concepts
+
+Some core concepts in the world of TPMs
+
+ - cryptography:
+    - digest functions
+    - hash extension
+    - HMAC
+    - symmetric encryption
+    - asymmetric encryption and/or signature algorithms
+ - platform configuration registers (PCRs)
+ - cryptographic object naming
+ - immutability of object public areas
+ - key hierarchies
+ - key wrapping
+ - limited resources
+    - tickets
+    - resource management
+ - sessions
+ - authorization
+    - restricted cryptographic keys
+    - policies
+ - other object types
+    - non-volatile (NV) indexes
+ - attestation
+
+We'll assume reader familiarity with cryptography so we need not explain
+it.
```

Comment:
We'll assume reader familiarity with the *basics* of cryptographic keys and how encryption, decryption, signing and verification work with asymmetric and symmetric keys, eg: RSA.

Reply:
Good addition @iolivergithub